backend/helpers/analyticsRouteHandlerHelpers.js

```diff
@@ -362,6 +514,45 @@ const getGroupAnalyticsEffort = async (unitCode, year, period, groupNumber) => {
     return effortAnalyticsData;
 };
 
+const getGroupAnalyticsPreference = async (unitCode, year, period, groupNumber) => {
+    /**
+     * Given a group, obtains all information related to student preference
+     * to group matching
+     */
+
+    const preferenceAnalyticsData = {
+        "personality title": "Preference Selection",
+        "description":
+            "Based on the input of each student, our algorithm attempts to a" +
+            "ssign each student their most preferred group. This is an aggre" +
+            "gation of how many students were assigned to their nth preferre" +
+            "d selection.",
+        data: [],
+    };
+
+    /** const preferenceResults = await promiseBasedQuery("FILL THIS SQL CODE IN THANKS" [unitCode, year, period, groupNumber]) */
```

Comment:
SQL not complete?

Reply:
https://github.com/Monash-FIT3170/BaLance-Team-Forming-Dashboard/pull/147#discussion_r1716276152
Same issue